services/src/datasets/in_memory.rs

```diff
@@ -97,7 +99,14 @@ impl HashMapStorable for StaticMetaData<MockDataSetDataSourceLoadingInfo, Vector
 
 impl HashMapStorable for GdalMetaDataRegular {
     fn store(&self, id: InternalDataSetId, db: &mut HashMapDataSetDb) -> TypedResultDescriptor {
-        db.gdal_data_sets.insert(id, self.clone());
+        db.gdal_data_sets.insert(id, Box::new(self.clone()));
+        self.result_descriptor.clone().into()
+    }
+}
+
+impl HashMapStorable for GdalMetaDataStatic {
+    fn store(&self, id: InternalDataSetId, db: &mut HashMapDataSetDb) -> TypedResultDescriptor {
+        db.gdal_data_sets.insert(id, Box::new(self.clone()));
```

Comment:
Can't you implement it for everything with certain trait bounds? Looks the same for me.

Reply:
I tried
```
impl<G> HashMapStorable for G
where
    G: MetaData<GdalLoadingInfo, RasterResultDescriptor>,
{
    fn store(&self, id: InternalDataSetId, db: &mut HashMapDataSetDb) -> TypedResultDescriptor {
        db.gdal_data_sets.insert(id, Box::new(self.clone()));
        self.result_descriptor.clone().into()
    }
}
```

but Rust wouldn't let me:
> error[E0119]: conflicting implementations of trait `datasets::in_memory::HashMapStorable` for type `geoengine_operators::engine::StaticMetaData<geoengine_operators::source::OgrSourceDataset, geoengine_operators::engine::VectorResultDescriptor>`:
   --> services/src/datasets/in_memory.rs:93:1
    |
86  |   impl HashMapStorable for StaticMetaData<OgrSourceDataset, VectorResultDescriptor> {
    |   --------------------------------------------------------------------------------- first implementation here
...
93  | / impl<G> HashMapStorable for G
94  | | where
95  | |     G: MetaData<GdalLoadingInfo, RasterResultDescriptor>,
96  | | {
...   |
100 | |     }
101 | | }
    | |_^ conflicting implementation for `geoengine_operators::engine::StaticMetaData<geoengine_operators::source::OgrSourceDataset, geoengine_operators::engine::VectorResultDescriptor>`
    |
    = note: upstream crates may add a new impl of trait `geoengine_operators::engine::MetaData<geoengine_operators::source::GdalLoadingInfo, geoengine_operators::engine::RasterResultDescriptor>` for type `geoengine_operators::engine::StaticMetaData<geoengine_operators::source::OgrSourceDataset, geoengine_operators::engine::VectorResultDescriptor>` in future versions